gocourse04/main.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import "fmt"
+
+// Наприклад, зоопарк ділиться на території: копитні, пернаті, примати… та інше.
+
+const (
+	Ungulates AnimalType = "Ungulates"
+	Feathered AnimalType = "Feathered"
+	Mammals   AnimalType = "Mammals"
+	Horses    Breed      = "Horses"
+	Pony      Breed      = "Pony"
+	Ducks     Breed      = "Ducks"
+	Warthoag  Breed      = "Warthoag"
+	Meerkat   Breed      = "Meerkat"
+)
+
+func main() {
+	fmt.Println("Start")
+	zoo := NewZoo()
+	fmt.Println(zoo)
+}
+
+type Areas map[AnimalType]Area
+
+type Area struct {
+	Number  int
+	Sectors map[Breed]Sector
+}
+
+type Sector struct {
+	Subtype Breed
+	Animals *[]Animal
```

Comment:
Чи можемо змінити тип з `*[]Animal` на `[]Animal`? Здається, так простіше код стане.

Reply:
Теж не виправлено.

Reply:
Не виправлено.

Reply:
з цією правкою програма перестає працювать, сходу падають тести, причина в тому що не добавляються елемети в слайс

Reply:
Програма має працювати, якщо видалити `*`:
```suggestion
	Animals []Animal
```

Потрібно трішки підправити функції, які працюють з цим слайсом звірів.

Пропоную перечитати статті для кращого розуміння слайсів та масивів:

- https://go.dev/blog/slices-intro
- https://go.dev/blog/slices
- https://go.dev/doc/effective_go#arrays
- https://go.dev/doc/effective_go#slices
- https://victoriametrics.com/blog/go-array/
- https://victoriametrics.com/blog/go-slice/
- https://sazak.io/articles/visual-guide-to-slices-in-go-2024-03-25
- https://quii.gitbook.io/learn-go-with-tests/go-fundamentals/arrays-and-slices